spec/classes/init_spec.rb

```diff
@@ -794,6 +823,41 @@
           it { should contain_class('pam') }
         end
       end
+
+      context "with pam_d_sshd_template set to pam/sshd.custom.erb without specifying pam_sshd_[auth|account|password|session]_lines" do
+        let :facts do
+          { :osfamily => v[:osfamily],
+            :"#{v[:releasetype]}" => v[:release],
+            :lsbdistid => v[:lsbdistid],
+          }
+        end
+        let(:params) { { :pam_d_sshd_template => 'pam/sshd.custom.erb' } }
+
+        it "should fail" do
+            expect {
+              should contain_class('pam')
+            }.to raise_error(Puppet::Error, /pam_sshd_\[auth|account|password|session\]_lines required for template pam\/sshd.custom.erb/)
+        end
+      end
+
+      [ :pam_sshd_auth_lines, :pam_sshd_account_lines, :pam_sshd_password_lines, :pam_sshd_session_lines ].each do |param|
+        context "with #{param} specified and pam_d_sshd_template not specified on #{v[:osfamily]} with #{v[:releasetype]} #{v[:release]}" do
```

Comment:
Looking at the docs [1] [2] this should work as expected. I have used something similar [3] in my AFS module with no reported errors so far.

We could add spec tests to proof the expected behaviour.

[1] https://docs.puppet.com/puppet/latest/reference/lang_conditional.html#behavior
[2] https://docs.puppet.com/puppet/latest/reference/lang_expressions.html#or
[3] https://github.com/Phil-Friderici/puppet-module-afs/blob/v1.2.6/manifests/init.pp#L90-L100